source/ways-of-working/environments/external-access.html.md.erb

```diff
@@ -0,0 +1,102 @@
+
+[Traefik](https://traefik.io/traefik/) is used as the ingress controller in HMCTS applications.
+
+The following traefik instances are available to use:
+
+| IngressClass | Default | Application configuration |
+|-|-|-|
+| traefik | Yes | Application will be reachable externally without VPN. End user will be required to authenticate to Azure AD before being redirected to the application |
+| traefik-private | No | Application will be available over VPN only |
+| traefik-no-proxy | No | Application will be reachable externally without VPN or authentication |
+
+### Protecting applications with Azure AD authentication
+    
+To ensure end users must authenticate to Azure AD before they can view the application in their web browser, you must also set the `enableOAuth` flag to `true`
+
+```yaml
+nodejs:
+    ingressClass: traefik
+    disableTraefikTls: false
+    enableOAuth: true
+```
+
+*In SDS, if you do not set an ingressClass flag in code, it will default to traefik.*
+
+#### Adding a redirect URI
+
+The authentication with Azure AD is handled by an Azure AD App Registration in the [HMCTS Demo Azure AD tenant](https://portal.azure.com/hmctsexecdemo.onmicrosoft.com). To allow end users to be authenticated to your application, you must add a redirect URI for your application to the Azure AD App Registration.
+
+*If you are not currently a member of the HMCTS Demo Azure AD tenant, check with your team to see if anyone else is. Everyone in this tenant has permission to invite others.*
+
+There are separate Azure AD App Registrations for CFT and SDS:
+
+- [CFT](https://portal.azure.com/hmctsexecdemo.onmicrosoft.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Authentication/appId/fbf4cb6e-f09d-4fc4-89fa-b94cb582cb18/isMSAApp/)
+- [SDS](https://portal.azure.com/hmctsexecdemo.onmicrosoft.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Authentication/appId/2a3f8b5a-ec0e-470d-b979-fc79d3e74cba/isMSAApp/)
+
+##### Azure Portal
+
+Navigate to the appropriate app registration, click on `Authentication` and click `Add URI` under `Redirect URIs`.
+
+Your redirect URI should be the full URI of your application appended with `/oauth-proxy/callback` e.g. https://toffee.demo.platform.hmcts.net/oauth-proxy/callback
+
+##### Azure CLI
+
+Sign into the HMCTS Demo tenant with your HMCTS account:
+
+```az login --tenant hmctsexecdemo.onmicrosoft.com --allow-no-subscriptions```
+
+Get the application id of the appropriate app registration by clicking on the [links](https://hmcts.github.io/ways-of-working/environments/external-access.html#adding-a-redirect-uri) or run the following Azure CLI command:
+
+```az ad app list --display-name "<APP REGISTRATION NAME>" --query '[].appId'```
+
+Update the reply-urls of the app registration. You must specify the existing reply-urls and append your additions. Each reply-url should be separated by a space.
```

Comment:
If you're going to add a CLI command that requires this sort of thing it should be an entire script that automates it.

i.e. the script should retrieve the existing redirect urls, and append the new one.

handling differences between CFT and SDS, you can maybe use the CFT / SDS switcher from the troubleshooting page.

_Note: When I asked for CLI command that was actually for inviting guests_

Reply:
Ah ok, I thought you meant updating the reply-urls. I've removed the CLI part for that since the GUI is probably easier and it's going to be automated anyway. There was a request to enable the ability to invite guests via Azure CLI or Azure PowerShell but this is not available as of yet.